src/hooks/use-toggle-state/use-toggle-state.js

```diff
@@ -1,12 +1,21 @@
-import React from 'react';
+import { useCallback, useState } from 'react';
 
 const useToggleState = (defaultValue = true) => {
-  const [isToggled, setIsToggled] = React.useState(defaultValue);
-  const toggle = React.useCallback(() => {
+  const [isToggled, setIsToggled] = useState(defaultValue);
+
+  const setOn = useCallback(() => {
+    setIsToggled(true);
+  }, [setIsToggled]);
+
+  const setOff = useCallback(() => {
+    setIsToggled(false);
+  }, [setIsToggled]);
+
+  const toggle = useCallback(() => {
     setIsToggled(!isToggled);
-  }, [isToggled]);
+  }, [isToggled, setIsToggled]);
 
-  return [isToggled, toggle];
+  return [isToggled, toggle, setOn, setOff];
```

Comment:
Hmm I'm not sure I like this new API.

Can't we just accept an argument to the `toggle` function, which takes precedence over the `!isToggled`?

```js
const toggle = useCallback((forceIsToggled) => {
  setIsToggled(typeof forceIsToggled === 'boolean' ? forceIsToggled : !isToggled);
}, [isToggled]);
```

Reply:
hm, cool idea 👍

Reply:
Hm. I don't like that force boolean arg. I like those tiny functions.

Reply:
this is the result of the force bool arg [here](https://github.com/commercetools/ui-kit/pull/984/commits/c56b5355d6672f95e2421a938ee983ba617d6d5b)

Reply:
Yeah I also prefer to have one setter and therefore one tuple as the return value. Having additional values feels fragile to me and it's really not clear that those functions are available.

I also assume that forcing the state is not something we would do regularly, so I'm fine with this being the exception.

Reply:
I got the original idea to add the isOn and isOff functions from this [hook](https://github.com/sgenoud/quarolib/blob/master/hooks/useToggle.js)

... because I was looking at his `usePopper` hook 😆

Reply:
I also like having the smol fns, but I agree it might be better to keep it simple with just a tuple.
 -  -  -

> this way I need to remember the order of the setOn and setOff functions

Alternatively, we could also make it return an object instead of an array, but its a bit less convenient for naming the fns.